client/network/src/service.rs

```diff
@@ -1534,6 +1527,13 @@ where
 
 				self.event_streams.send(Event::Dht(event));
 			},
+			SwarmEvent::Behaviour(BehaviourOut::AutoNAT(event)) => {
```

Comment:
Is `AutoNAT` protocol the one responsible for `ExternalAddressConfirmed` events?

Reply:
Yes.